### PR TITLE
fixes #6961

### LINKED ISSUE #6961: "Stay alive until the end" objective failed while alive
Revision: 666678ecda4be02cff456ff778ab632ad8c705e7 

> General description of the issue

I failed my "Stay alive until the end" objective as a AutoTraitor'd AI (made traitor during the round), even though I was alive. I had never been damaged. I was made an AI during the round.
stay alive until the end
As it shows, I survived, but the objective success did not trigger.

> What you expected to happen

Sweet, sweet greentext.

> What actually happened

My other objective - kill the HoS - was completed, but the "stay alive" objective was not.

> Steps to reproduce if possible.

Become a traitor AI (mid-round?) with the "stay alive until the end" objective and stay alive until the end.

### PR DESCRIPTION
**Pull request by Intigracy**
sillycons now have a survive objective suitable to them

fixes #6961

### CODE REVIEW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -523,7 +522,10 @@ var/list/potential_theft_objectives=list(
 /datum/objective/nuclear
 	explanation_text = "Destroy the station with a nuclear device."
 
-
+	check_completion()
+		if(ticker.mode.explosion_in_progress || ticker.mode.station_was_nuked)
+			return 1
+		return 0
```
Comment (Intigracy) [inline code/game/gamemodes/objective.dm]: nonatomic bugfix, nuke station objective doesn't even check completion currently

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -523,7 +522,10 @@ var/list/potential_theft_objectives=list(
 /datum/objective/nuclear
 	explanation_text = "Destroy the station with a nuclear device."
 
-
+	check_completion()
```
Comment (PJB3005) [inline code/game/gamemodes/objective.dm]: Wow relative pathing.

File: code/game/gamemodes/objective.dm [hunk 3]
```diff
@@ -407,20 +403,23 @@ var/list/potential_theft_objectives=list(
...
+		if(!owner.current || owner.current.stat == DEAD || isbrain(owner.current) || isborer(owner.current || issilicon(owner.current)))
 			return 0		//Brains no longer win survive objectives. --NEO
-		if(issilicon(owner.current) && owner.current != owner.original)
-			return 0
 		return 1
 
+/datum/objective/siliconsurvive
+	explanation_text = "Remain functional until the end."
 
+	check_completion()
```
Comment (PJB3005) [inline code/game/gamemodes/objective.dm]: Wow

Comment (Intigracy) [inline code/game/gamemodes/objective.dm]: do you want me to fix this entire file then?

**Comment by Intigracy** [comment]
done

**Pull request merged**